core_lib_generator/core_lib_template_generator.py

```diff
@@ -0,0 +1,235 @@
+                print('\nSQLite created in memory.')
+                add_db = input_yes_no('\nDo you want to add another DB connection?', False)
+                db_template[db_name.lower()] = {
+                    'log_queries': db_log_queries,
+                    'create_db': db_create,
+                    'session': {
+                        'pool_recycle': db_pool_recycle,
+                        'pool_pre_ping': db_pool_pre_ping,
+                    },
+                    'url': {
+                        'protocol': DBTypes(db_type).name.lower(),
+                    },
+                }
+                continue
+        default_db_ports = {
+            DBTypes.SQLite.name: None,
+            DBTypes.Postgresql.name: 5432,
+            DBTypes.MySQL.name: 3306,
+            DBTypes.Oracle.name: 1521,
+            DBTypes.MSSQL.name: 1433,
+            DBTypes.Firebird.name: 3050,
+            DBTypes.Sybase.name: 5000,
+        }
+        db_port = input_int('Enter the port no. of your DB', default_db_ports[DBTypes(db_type).name])
+        db_host = input_str('Enter host of your DB', 'localhost')
+        db_username = input_str('Enter your DB username', 'user')
+        db_password = input_str('Enter your DB password', 'password')
+        print(f'\n{DBTypes(db_type).name} with {db_host}:{db_port} created')
+        add_db = input_yes_no('\nDo you want to add another DB connection?', False)
+        db_template[db_name.lower()] = {
+            'env': {
+                f'{db_name.upper()}_USER': db_username,
+                f'{db_name.upper()}_PASSWORD': db_password,
+                f'{db_name.upper()}_PORT': db_port,
+                f'{db_name.upper()}_DB': db_name,
+                f'{db_name.upper()}_HOST': db_host,
+            },
+            'config': {
+                'protocol': DBTypes(db_type).name.lower(),
+                'username': f'${{oc.env:{db_name.upper()}_USER}}',
+                'password': f'${{oc.env:{db_name.upper()}_PASSWORD}}',
+                'host': f'${{oc.env:{db_name.upper()}_HOST}}',
+                'port': f'${{oc.env:{db_name.upper()}_PORT}}',
+                'file': f'${{oc.env:{db_name.upper()}_DB}}',
+            },
+        }
+    return db_template
+
+
+def generate_solr_template() -> dict:
+    solr_protocol = input_str('Enter solr protocol (http/https)', 'https')
+    solr_port = input_int('Enter solr port no.', 8983)
+    solr_host = input_str('Enter solr host', 'localhost')
+    solr_file = input_str('Enter solr file name (solr/core_name)', 'solr/mycore')
+    print('\nInput recorded.')
+    return {
+        'env': {
+            'SOLR_HOST': solr_host,
+            'SOLR_PORT': solr_port,
+        },
+        'config': {
+            'protocol': solr_protocol,
+            'host': f'${{oc.env:SOLR_HOST}}',
+            'port': f'${{oc.env:SOLR_PORT}}',
+            'file': solr_file,

```

Comment:
solr have no file name in the connection. and it's http predefined for the protocol

Reply:
I had seen it in one of your configs so included it, this is for the `core` name actually